### ISSUE TITLE
--with-wolfssl configurations is broken

### ISSUE THREAD

**Issue by tmael** [chriskohlhoff/asio#574]
Hi @chriskohlhoff, 

A build --with-wolfssl configuration option is broken.

wolfSSL previously submitted PR #383 which has been merged into the asio mainline. Thank you! 

Unfortunately, the --with-wolfssl build option hook in configure.ac was missed when cherry-picking the changes. 

Can you please merge the pieces that were missed? 

The workaround is to build with the following options.
./configure CPPFLAGS="-DASIO_USE_WOLFSSL" LDFLAGS="-lwolfssl"

Please let me know if you need more info or if you would like me to submit a new pull request. 

Thanks,
Tesfa

**Comment by tomoveu** [comment]
I can confirm that wolfssl option switch is broken. 

![asio-wolfssl-switch-broken](https://user-images.githubusercontent.com/11628321/101179680-3be54a00-3653-11eb-90ce-531dae16f15a.png)

**Comment by tomoveu** [comment]
Hi @chriskohlhoff could you please comment. 

The original upstreaming effort was good, but the partial merge(cherry pick) caused a broken configuration.

Thanks,
Dimi

**Comment by tmael** [comment]
Hi Team ASIO, 

This issue is blocking users of ASIO with wolfSSL building against the main-line for production releases.
Any update on this issue? 
Please let me know. 

Thanks,
Tesfa

**Comment by ppavlik** [comment]
@tmael , can I suggest you create a PR for this issue to help move things along? This has been outstanding for too long.

Thanks,
Paul

**Comment by tmael** [comment]
@ppavlik, good idea! haven't had luck getting an email response from Chris neither. Anyway, @dgarske put up a PR to address the issue. Hoping it gets merged quickly.  Thanks
